Nitrox.Test/Model/Packets/PacketsSerializableTest.cs

```diff
@@ -1,95 +1,94 @@
-                surrogatedType.Should().NotBeNull();
-                surrogateSelector.AddSurrogate(surrogatedType, streamingContext, surrogate);
-            }
+                testedType = type;
 
-            // For completeness, we could pass a StreamingContextStates.CrossComputer.
-            BinaryFormatter serializer = new BinaryFormatter(surrogateSelector, streamingContext);
+                object faker = typeof(NitroxAutoFaker<>).MakeGenericType(type)
+                    .GetConstructor(new Type[] { typeof(Dictionary<Type, Type[]>) })
+                    .Invoke(new object[] { subtypesByBaseType.ToDictionary(pair => pair.Key, pair => pair.Value) });
 
-            Stream stream = new MemoryStream();
+                if (subtypesByBaseType.ContainsKey(type))
+                {
+                    for (int i = 0; i < subtypesByBaseType[type].Length; i++)
+                    {
+                        Packet packet = (Packet)typeof(AutoFaker<>).MakeGenericType(subtypesByBaseType[type][i])
+                        .GetMethod(nameof(AutoFaker<object>.Generate), new Type[] { typeof(string) })
+                        .Invoke(faker, new object[] { null });
+                        packets.Add(packet);
+                    }
+                }
+                else
+                {
+                    Packet packet = (Packet)typeof(AutoFaker<>).MakeGenericType(type)
+                    .GetMethod(nameof(AutoFaker<object>.Generate), new Type[] { typeof(string) })
+                    .Invoke(faker, new object[] { null });
+                    packets.Add(packet);
+                }
+            }
 
-            Type testedType = null;
-            List<Packet> packets = new List<Packet>();
+            Packet.InitSerializer();
+            bool failed = false;
 
-            foreach (Type type in typeof(Packet).Assembly.GetTypes()
-            .Where(p => typeof(Packet).IsAssignableFrom(p) && p.IsClass && !p.IsAbstract))
-            {
-                testedType = type;
-                packets.Add((Packet)FormatterServices.GetUninitializedObject(type));
-            }
+            CompareLogic logic = new();
+            logic.Config.SkipInvalidIndexers = true;
+            logic.Config.AttributesToIgnore.Add(typeof(IgnoredMemberAttribute));
 
             foreach (Packet packet in packets)
             {
                 testedType = packet.GetType();
-                serializer.Serialize(stream, packet);
-                stream.Flush();
-                stream.Position = 0;
-                serializer.Deserialize(stream);
-                stream.Position = 0;
+                Packet deserialized = Packet.Deserialize(packet.Serialize());
+
+                ComparisonResult result = logic.Compare(packet, deserialized);
+                if (!result.AreEqual)
+                {
+                    Console.WriteLine($"Differences found:\n{result.DifferencesString}");
```

Comment:
This will stop the test as soon as a difference is found. It is better to wait until all packets have been compared before throwing an error in order to catch multiple errors at once.